lib/nerve/utils.rb

```diff
@@ -4,5 +4,14 @@ def safe_run(command)
       res = `#{command}`.chomp
       raise "command '#{command}' failed to run:\n#{res}" unless $?.success?
     end
+
+    def responsive_sleep(seconds)
+      nap_time = seconds
+      while nap_time > 0
+        yield if block_given?
+        sleep [nap_time, 1].min
```

Comment:
oh. i didn't think about it. yeah, so, probably a better way would be to declare that the block you pass to `responsive_sleep` returns a value which determines if the sleep should terminate. e.g.:

```ruby
def responsive_sleep(seconds, &should_we_exit)
  while true
    sleep 1
    break if (should_we_exit && should_we_exit.call)
  end
end
```